types/wordpress__editor/store/selectors.d.ts

```diff
@@ -137,9 +141,14 @@ export function getCurrentPostType(): string;
  *
  * @param attributeName - Post attribute name.
  */
-export function getEditedPostAttribute<T extends keyof (Page | Post)>(
-    attributeName: T
-): (Page | Post)[T] | undefined;
+export interface getEditedPostAttribute extends SelectorWithCustomCurrySignature {
+    <T extends keyof (Page | Post)>(
+        state: StateType,
+        attributeName: T
+      ): (Page | Post)[T] | undefined;
+
+    CurriedSignature: <T extends keyof (Page | Post)>(attributeName: T) => (Page | Post)[T] | undefined;
+}
```

Comment:
This doesn't work, so I need to get feedback on it. See this (https://github.com/WordPress/gutenberg/pull/41578) for a huge description about why it's needed.

The TLDR is that complex generics (e.g. `T extends key of Post`, returning `Post[T]`) don't get carried over when wp data's internal types remove the `state` argument. As a result, type inference is very broad -- for example `getEditedPostAttribute( 'author' )` returns `number|undefined`. This works previously, and if you were to call the selector on its own... but with the new type inference, the return type is extraordinarily broad (basically the full set of Page|Post value types.) 

The fix is this custom interface which _should_ make the internal types shortcut and just use the given signature manually instead of inferring it... but this isn't working for some reason.